### PR TITLE
Add a fixed order for elements in exported CNE (#12.6)

### PR DESCRIPTION
**Pull request by pet-mit**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Feature

**What is the current behavior?** *(You can also link to an open issue here)*
Elements in the exported CNE are sorted randomly

**What is the new behavior (if this is a feature change)?**
Sorting rules have been implemented. Thus the CNE export is replicable

### CODE REVIEW

File: data/crac-io/crac-io-cne/src/main/java/com/farao_community/farao/data/crac_io_cne/Cne.java [hunk 1]
```diff
@@ -94,11 +98,20 @@ private void createAllConstraintSeries(Point point) {
         Crac crac = cneHelper.getCrac();
 
         List<ConstraintSeries> constraintSeriesList = new ArrayList<>();
-        crac.getBranchCnecs().forEach(cnec -> createConstraintSeriesOfACnec(cnec, cneHelper, constraintSeriesList));
+        List<BranchCnec> sortedCnecs = new ArrayList<>(crac.getBranchCnecs());
+        sortedCnecs.sort(Comparator.comparing(Identifiable::getId));
+        sortedCnecs.forEach(cnec -> createConstraintSeriesOfACnec(cnec, cneHelper, constraintSeriesList));
```
Comment (TheMaskedTurtle) [inline data/crac-io/crac-io-cne/src/main/java/com/farao_community/farao/data/crac_io_cne/Cne.java:103]: I think this is more compact `personList.stream().sorted(Comparator.comparing(Person::getPersonId)).forEach(person -> System.out.println(person.getName()));`

File: data/crac-io/crac-io-cne/src/main/java/com/farao_community/farao/data/crac_io_cne/Cne.java [hunk 2]
```diff
@@ -94,11 +98,20 @@ private void createAllConstraintSeries(Point point) {
...
 
         ConstraintSeries preventiveB56 = newConstraintSeries(generateRandomMRID(), B56_BUSINESS_TYPE);
-        crac.getRangeActions().forEach(rangeAction -> createRangeRemedialActionSeries(rangeAction, cneHelper, constraintSeriesList, preventiveB56));
-        crac.getNetworkActions().forEach(networkAction -> createNetworkRemedialActionSeries(networkAction, cneHelper, preventiveB56));
+
+        List<RangeAction> sortedRangeActions = new ArrayList<>(crac.getRangeActions());
+        sortedRangeActions.sort(Comparator.comparing(Identifiable::getId));
+        sortedRangeActions.forEach(rangeAction -> createRangeRemedialActionSeries(rangeAction, cneHelper, constraintSeriesList, preventiveB56));
```
Comment (TheMaskedTurtle) [inline data/crac-io/crac-io-cne/src/main/java/com/farao_community/farao/data/crac_io_cne/Cne.java:109]: same

File: data/crac-io/crac-io-cne/src/main/java/com/farao_community/farao/data/crac_io_cne/Cne.java [hunk 3]
```diff
@@ -94,11 +98,20 @@ private void createAllConstraintSeries(Point point) {
...
+
+        List<NetworkAction> sortedNetworkActions = new ArrayList<>(crac.getNetworkActions());
+        sortedNetworkActions.sort(Comparator.comparing(Identifiable::getId));
+        sortedNetworkActions.forEach(networkAction -> createNetworkRemedialActionSeries(networkAction, cneHelper, preventiveB56));
```
Comment (TheMaskedTurtle) [inline data/crac-io/crac-io-cne/src/main/java/com/farao_community/farao/data/crac_io_cne/Cne.java:113]: same

File: data/crac-io/crac-io-cne/src/main/java/com/farao_community/farao/data/crac_io_cne/CneClassCreator.java [hunk 4]
```diff
@@ -93,7 +90,9 @@ public static ConstraintSeries newConstraintSeries(String id, String businessTyp
     public static ConstraintSeries newConstraintSeries(String id, String businessType, Set<Country> countries, String optimStatus) {
         ConstraintSeries constraintSeries = newConstraintSeries(id, businessType, optimStatus);
         if (!countries.isEmpty()) {
-            countries.forEach(country -> constraintSeries.partyMarketParticipant.add(newPartyMarketParticipant(country)));
+            List<Country> sortedCountries = new ArrayList<>(countries);
+            sortedCountries.sort(Comparator.comparing((Country c) -> new EICode(c).getCode()));
+            sortedCountries.forEach(country -> constraintSeries.partyMarketParticipant.add(newPartyMarketParticipant(country)));
```
Comment (TheMaskedTurtle) [inline data/crac-io/crac-io-cne/src/main/java/com/farao_community/farao/data/crac_io_cne/CneClassCreator.java:95]: same

File: data/crac-io/crac-io-cne/src/main/java/com/farao_community/farao/data/crac_io_cne/CneCnecsCreator.java [hunk 5]
```diff
@@ -50,9 +50,10 @@ static void createConstraintSeriesOfACnec(BranchCnec cnec, CneHelper cneHelper,
         Set<Country> countries = createCountries(network, cnec.getNetworkElement().getId());
 
         /* Create Constraint series */
-        ConstraintSeries constraintSeriesB54 = newConstraintSeries(cnec.getId(), B54_BUSINESS_TYPE, countries, OPTIMIZED_MARKET_STATUS);
-        ConstraintSeries constraintSeriesB57 = newConstraintSeries(cnec.getId(), B57_BUSINESS_TYPE, countries, OPTIMIZED_MARKET_STATUS);
-        ConstraintSeries constraintSeriesB88 = newConstraintSeries(cnec.getId(), B88_BUSINESS_TYPE, countries, OPTIMIZED_MARKET_STATUS);
+        String marketStatus = (cnec.isOptimized() || !cnec.isMonitored()) ? OPTIMIZED_MARKET_STATUS : MONITORED_MARKET_STATUS; // TO DO : separate CNECs from MNECs
```
Comment (TheMaskedTurtle) [inline data/crac-io/crac-io-cne/src/main/java/com/farao_community/farao/data/crac_io_cne/CneCnecsCreator.java:53]: Do you really need to specify `|| !cnec.isMonitored()` ?

**Review by baptsegu — DISMISSED** [review]

**Review by TheMaskedTurtle — APPROVED** [review]

**Pull request merged**